File: packages/flutterfire_gen/test/helper/app_user_post_like.flutterfire_gen.dart
// coverage:ignore-file
// ignore_for_file: type=lint
// ignore_for_file: unused_element, deprecated_member_use, deprecated_member_use_from_same_package, use_function_type_syntax_for_parameters, unnecessary_const, avoid_init_to_null, invalid_override_different_default_values_named, prefer_expression_function_bodies, annotate_overrides, invalid_annotation_target, unnecessary_question_mark

part of 'app_user_post_like.dart';

class ReadAppUserPostLike {
  const ReadAppUserPostLike({
    required this.appUserPostLikeId,
    required this.path,
    required this.likedByAppUserId,
    required this.likedAt,
    required this.updatedAt,
  });

  final String appUserPostLikeId;

  final String path;

  final String likedByAppUserId;

  final DateTime? likedAt;

  final DateTime? updatedAt;

  factory ReadAppUserPostLike._fromJson(Map<String, dynamic> json) {
    final extendedJson = <String, dynamic>{
      ...json,
    };
    return ReadAppUserPostLike(
      appUserPostLikeId: extendedJson['appUserPostLikeId'] as String,
      path: extendedJson['path'] as String,
      likedByAppUserId: extendedJson['likedByAppUserId'] as String,
      likedAt: (extendedJson['likedAt'] as Timestamp?)?.toDate(),
      updatedAt: (extendedJson['updatedAt'] as Timestamp?)?.toDate(),
    );
  }

  factory ReadAppUserPostLike.fromDocumentSnapshot(DocumentSnapshot ds) {
    final data = ds.data()! as Map<String, dynamic>;
    return ReadAppUserPostLike._fromJson(<String, dynamic>{
      ...data,
      'appUserPostLikeId': ds.id,
      'path': ds.reference.path,
    });
  }
}

class CreateAppUserPostLike {
  const CreateAppUserPostLike({
    required this.likedByAppUserId,
  });

  final String likedByAppUserId;

  Map<String, dynamic> toJson() {
    final json = <String, dynamic>{
      'likedByAppUserId': likedByAppUserId,
      'likedAt': FieldValue.serverTimestamp(),
      'updatedAt': FieldValue.serverTimestamp(),
    };
    final jsonPostProcessors = <({String key, dynamic value})>[];
    return {
      ...json,
      ...Map.fromEntries(jsonPostProcessors
          .map((record) => MapEntry(record.key, record.value))),
    };
  }
}

class UpdateAppUserPostLike {
  const UpdateAppUserPostLike({
    this.likedByAppUserId,
    this.likedAt,
  });

  final String? likedByAppUserId;
  final DateTime? likedAt;

  Map<String, dynamic> toJson() {
    final json = <String, dynamic>{
      if (likedByAppUserId != null) 'likedByAppUserId': likedByAppUserId,
      if (likedAt != null) 'likedAt': likedAt,
      'updatedAt': FieldValue.serverTimestamp(),
    };
    final jsonPostProcessors = <({String key, dynamic value})>[];
    return {
      ...json,
      ...Map.fromEntries(jsonPostProcessors
          .map((record) => MapEntry(record.key, record.value))),
    };
  }
}

class DeleteAppUserPostLike {}

/// Provides a reference to the appUserPostLikes collection for reading.
CollectionReference<ReadAppUserPostLike>
    readAppUserPostLikeCollectionReference({
  required String appUserId,
  required String appUserPostId,
}) =>
        FirebaseFirestore.instance
            .collection('appUsers')
            .doc(appUserId)
            .collection('appUserPosts')
            .doc(appUserPostId)
            .collection('appUserPostLikes')
            .withConverter<ReadAppUserPostLike>(
              fromFirestore: (ds, _) =>
                  ReadAppUserPostLike.fromDocumentSnapshot(ds),
              toFirestore: (_, __) => throw UnimplementedError(),
            );

/// Provides a reference to a appUserPostLike document for reading.
DocumentReference<ReadAppUserPostLike> readAppUserPostLikeDocumentReference({
  required String appUserId,
  required String appUserPostId,
  required String appUserPostLikeId,
}) =>
    readAppUserPostLikeCollectionReference(
            appUserId: appUserId, appUserPostId: appUserPostId)
        .doc(appUserPostLikeId);

/// Provides a reference to the appUserPostLikes collection for creating.
CollectionReference<CreateAppUserPostLike>
    createAppUserPostLikeCollectionReference({
  required String appUserId,
  required String appUserPostId,
}) =>
        FirebaseFirestore.instance
            .collection('appUsers')
            .doc(appUserId)
            .collection('appUserPosts')
            .doc(appUserPostId)
            .collection('appUserPostLikes')
            .withConverter<CreateAppUserPostLike>(
              fromFirestore: (_, __) => throw UnimplementedError(),
              toFirestore: (obj, _) => obj.toJson(),
            );

/// Provides a reference to a appUserPostLike document for creating.
DocumentReference<CreateAppUserPostLike>
    createAppUserPostLikeDocumentReference({
  required String appUserId,
  required String appUserPostId,
  required String appUserPostLikeId,
}) =>
        createAppUserPostLikeCollectionReference(
                appUserId: appUserId, appUserPostId: appUserPostId)
            .doc(appUserPostLikeId);

/// Provides a reference to the appUserPostLikes collection for updating.
CollectionReference<UpdateAppUserPostLike>
    updateAppUserPostLikeCollectionReference({
  required String appUserId,
  required String appUserPostId,
}) =>
        FirebaseFirestore.instance
            .collection('appUsers')
            .doc(appUserId)
            .collection('appUserPosts')
            .doc(appUserPostId)
            .collection('appUserPostLikes')
            .withConverter<UpdateAppUserPostLike>(
              fromFirestore: (_, __) => throw UnimplementedError(),
              toFirestore: (obj, _) => obj.toJson(),
            );

/// Provides a reference to a appUserPostLike document for updating.
DocumentReference<UpdateAppUserPostLike>
    updateAppUserPostLikeDocumentReference({
  required String appUserId,
  required String appUserPostId,
  required String appUserPostLikeId,
}) =>
        updateAppUserPostLikeCollectionReference(
                appUserId: appUserId, appUserPostId: appUserPostId)
            .doc(appUserPostLikeId);

/// Provides a reference to the appUserPostLikes collection for deleting.
CollectionReference<DeleteAppUserPostLike>
    deleteAppUserPostLikeCollectionReference({
  required String appUserId,
  required String appUserPostId,
}) =>
        FirebaseFirestore.instance
            .collection('appUsers')
            .doc(appUserId)
            .collection('appUserPosts')
            .doc(appUserPostId)
            .collection('appUserPostLikes')
            .withConverter<DeleteAppUserPostLike>(
              fromFirestore: (_, __) => throw UnimplementedError(),
              toFirestore: (_, __) => throw UnimplementedError(),
            );

/// Provides a reference to a appUserPostLike document for deleting.
DocumentReference<DeleteAppUserPostLike>
    deleteAppUserPostLikeDocumentReference({
  required String appUserId,
  required String appUserPostId,
  required String appUserPostLikeId,
}) =>
        deleteAppUserPostLikeCollectionReference(
                appUserId: appUserId, appUserPostId: appUserPostId)
            .doc(appUserPostLikeId);

/// Manages queries against the appUserPostLikes collection.
class AppUserPostLikeQuery {
  /// Fetches [ReadAppUserPostLike] documents.
  Future<List<ReadAppUserPostLike>> fetchDocuments({
    required String appUserId,
    required String appUserPostId,
    GetOptions? options,
    Query<ReadAppUserPostLike>? Function(Query<ReadAppUserPostLike> query)?
        queryBuilder,
    int Function(ReadAppUserPostLike lhs, ReadAppUserPostLike rhs)? compare,
  }) async {
    Query<ReadAppUserPostLike> query = readAppUserPostLikeCollectionReference(
        appUserId: appUserId, appUserPostId: appUserPostId);
    if (queryBuilder != null) {
      query = queryBuilder(query)!;
    }
    final qs = await query.get(options);
    final result = qs.docs.map((qds) => qds.data()).toList();
    if (compare != null) {
      result.sort(compare);
    }
    return result;
  }

  /// Subscribes [AppUserPostLike] documents.
  Stream<List<ReadAppUserPostLike>> subscribeDocuments({
    required String appUserId,
    required String appUserPostId,
    Query<ReadAppUserPostLike>? Function(Query<ReadAppUserPostLike> query)?
        queryBuilder,
    int Function(ReadAppUserPostLike lhs, ReadAppUserPostLike rhs)? compare,
    bool includeMetadataChanges = false,
    bool excludePendingWrites = false,
  }) {
    Query<ReadAppUserPostLike> query = readAppUserPostLikeCollectionReference(
        appUserId: appUserId, appUserPostId: appUserPostId);
    if (queryBuilder != null) {
      query = queryBuilder(query)!;
    }
    var streamQs =
        query.snapshots(includeMetadataChanges: includeMetadataChanges);
    if (excludePendingWrites) {
      streamQs = streamQs.where((qs) => !qs.metadata.hasPendingWrites);
    }
    return streamQs.map((qs) {
      final result = qs.docs.map((qds) => qds.data()).toList();
      if (compare != null) {
        result.sort(compare);
      }
      return result;
    });
  }

  /// Fetches a specific [ReadAppUserPostLike] document.
  Future<ReadAppUserPostLike?> fetchDocument({
    required String appUserId,
    required String appUserPostId,
    required String appUserPostLikeId,
    GetOptions? options,
  }) async {
    final ds = await readAppUserPostLikeDocumentReference(
      appUserId: appUserId,
      appUserPostId: appUserPostId,
      appUserPostLikeId: appUserPostLikeId,
    ).get(options);
    return ds.data();
  }

  /// Subscribes a specific [AppUserPostLike] document.
  Stream<ReadAppUserPostLike?> subscribeDocument({
    required String appUserId,
    required String appUserPostId,
    required String appUserPostLikeId,
    bool includeMetadataChanges = false,
    bool excludePendingWrites = false,
  }) {
    var streamDs = readAppUserPostLikeDocumentReference(
      appUserId: appUserId,
      appUserPostId: appUserPostId,
      appUserPostLikeId: appUserPostLikeId,
    ).snapshots(includeMetadataChanges: includeMetadataChanges);
    if (excludePendingWrites) {
      streamDs = streamDs.where((ds) => !ds.metadata.hasPendingWrites);
    }
    return streamDs.map((ds) => ds.data());
  }

  /// Adds a [AppUserPostLike] document.
  Future<DocumentReference<CreateAppUserPostLike>> add({
    required String appUserId,
    required String appUserPostId,
    required CreateAppUserPostLike createAppUserPostLike,
  }) =>
      createAppUserPostLikeCollectionReference(
              appUserId: appUserId, appUserPostId: appUserPostId)
          .add(createAppUserPostLike);

  /// Sets a [AppUserPostLike] document.
  Future<void> set({
    required String appUserId,
    required String appUserPostId,
    required String appUserPostLikeId,
    required CreateAppUserPostLike createAppUserPostLike,
    SetOptions? options,
  }) =>
      createAppUserPostLikeDocumentReference(
        appUserId: appUserId,
        appUserPostId: appUserPostId,
        appUserPostLikeId: appUserPostLikeId,
      ).set(createAppUserPostLike, options);

  /// Updates a specific [AppUserPostLike] document.
  Future<void> update({
    required String appUserId,
    required String appUserPostId,
    required String appUserPostLikeId,
    required UpdateAppUserPostLike updateAppUserPostLike,
  }) =>
      updateAppUserPostLikeDocumentReference(
        appUserId: appUserId,
        appUserPostId: appUserPostId,
        appUserPostLikeId: appUserPostLikeId,
      ).update(updateAppUserPostLike.toJson());

  /// Deletes a specific [AppUserPostLike] document.
  Future<void> delete({
    required String appUserId,
    required String appUserPostId,
    required String appUserPostLikeId,
  }) =>
      deleteAppUserPostLikeDocumentReference(
        appUserId: appUserId,
        appUserPostId: appUserPostId,
        appUserPostLikeId: appUserPostLikeId,
      ).delete();
}

File: packages/flutterfire_gen_ts/test/helper/app_user_post_like.flutterfireGen.ts
import * as admin from 'firebase-admin'
import {
    CollectionReference,
DocumentData,
DocumentReference,
DocumentSnapshot,
Query,
QueryDocumentSnapshot,
QuerySnapshot,
SetOptions,
Timestamp,
WriteResult
} from 'firebase-admin/firestore'




export class ReadAppUserPostLike {
  constructor ({
    appUserPostLikeId,
    path,
    likedByAppUserId,
stringWithDefaultValue,
nullableString,
nullableStringWithDefaultValue,
likedAt,
  }: {
    appUserPostLikeId: string
    path: string
    likedByAppUserId: string
stringWithDefaultValue: string
nullableString?: string
nullableStringWithDefaultValue?: string
likedAt?: Date
  }) {
    this.appUserPostLikeId = appUserPostLikeId
    this.path = path
    this.likedByAppUserId = likedByAppUserId
this.stringWithDefaultValue = stringWithDefaultValue
this.nullableString = nullableString
this.nullableStringWithDefaultValue = nullableStringWithDefaultValue
this.likedAt = likedAt
  }

  readonly appUserPostLikeId: string

  readonly path: string

  readonly likedByAppUserId: string

readonly stringWithDefaultValue: string

readonly nullableString?: string

readonly nullableStringWithDefaultValue?: string

readonly likedAt?: Date

  

  private static fromJson(json: Record<string, unknown>): ReadAppUserPostLike {
    return new ReadAppUserPostLike({
      appUserPostLikeId: json['appUserPostLikeId'] as string,
      path: json['path'] as string,
      likedByAppUserId: json['likedByAppUserId'] as string,
stringWithDefaultValue: (json['stringWithDefaultValue'] as string | undefined) ?? 'defaultText',
nullableString: json['nullableString'] as string | undefined,
nullableStringWithDefaultValue: (json['nullableStringWithDefaultValue'] as string | undefined) ?? 'defaultText',
likedAt: (json['likedAt'] as Timestamp | undefined)?.toDate(),
    })
  }


  static fromDocumentSnapshot(
    ds: DocumentSnapshot
  ): ReadAppUserPostLike {
      const data = ds.data()!
      const cleanedData: Record<string, unknown> = {}
      for (const [key, value] of Object.entries(data)) {
          cleanedData[key] = value === null ? undefined : value
      }
      return ReadAppUserPostLike.fromJson({
          ...cleanedData,
          appUserPostLikeId: ds.id,
          path: ds.ref.path
      })
  }

}

export class CreateAppUserPostLike {
  constructor({
    likedByAppUserId,
stringWithDefaultValue,
nullableString,
nullableStringWithDefaultValue,
likedAt,
  }: {
    likedByAppUserId: string
stringWithDefaultValue: string
nullableString?: string
nullableStringWithDefaultValue?: string
likedAt?: Date
  }) {
    this.likedByAppUserId = likedByAppUserId
this.stringWithDefaultValue = stringWithDefaultValue
this.nullableString = nullableString
this.nullableStringWithDefaultValue = nullableStringWithDefaultValue
this.likedAt = likedAt
  }


  readonly likedByAppUserId: string

readonly stringWithDefaultValue: string

readonly nullableString?: string

readonly nullableStringWithDefaultValue?: string

readonly likedAt?: Date

  

  toJson(): Record<string, unknown> {
  return {
    likedByAppUserId: this.likedByAppUserId,
stringWithDefaultValue: this.stringWithDefaultValue,
nullableString: this.nullableString,
nullableStringWithDefaultValue: this.nullableStringWithDefaultValue,
likedAt: this.likedAt,
  };
}

}

export class UpdateAppUserPostLike {
  constructor({
    likedByAppUserId,
stringWithDefaultValue,
nullableString,
nullableStringWithDefaultValue,
likedAt,
  }: {
    likedByAppUserId?: string
stringWithDefaultValue?: string
nullableString?: string
nullableStringWithDefaultValue?: string
likedAt?: Date
  }) {
    this.likedByAppUserId = likedByAppUserId
this.stringWithDefaultValue = stringWithDefaultValue
this.nullableString = nullableString
this.nullableStringWithDefaultValue = nullableStringWithDefaultValue
this.likedAt = likedAt
  }


  readonly likedByAppUserId?: string

readonly stringWithDefaultValue?: string

readonly nullableString?: string

readonly nullableStringWithDefaultValue?: string

readonly likedAt?: Date

  

  // eslint-disable-next-line @typescript-eslint/no-explicit-any
toJson(): Record<string, any> {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const json: Record<string, any> = {}
  if (this.likedByAppUserId != undefined) {
  json['likedByAppUserId'] = this.likedByAppUserId
}
if (this.stringWithDefaultValue != undefined) {
  json['stringWithDefaultValue'] = this.stringWithDefaultValue
}
if (this.nullableString != undefined) {
  json['nullableString'] = this.nullableString
}
if (this.nullableStringWithDefaultValue != undefined) {
  json['nullableStringWithDefaultValue'] = this.nullableStringWithDefaultValue
}
if (this.likedAt != undefined) {
  json['likedAt'] = this.likedAt
}

  return json
}

}

export class DeleteAppUserPostLike {}

/**
 * A Cloud Firestore object which ignores `undefined` properties.
 */
const db = admin.firestore()
db.settings({ ignoreUndefinedProperties: true })

/**
 * Provides a reference to the appUserPostLikes collection for reading.
 */
export const readAppUserPostLikeCollectionReference = (
  {
  appUserId,
appUserPostId
}: {
  appUserId: string,
appUserPostId: string
}
): CollectionReference<ReadAppUserPostLike>
=> { return db
.collection('appUsers').doc(appUserId).collection('appUserPosts').doc(appUserPostId).collection('appUserPostLikes').withConverter<ReadAppUserPostLike>({
  fromFirestore: (ds: DocumentSnapshot): ReadAppUserPostLike => {
    return ReadAppUserPostLike.fromDocumentSnapshot(ds)
  },
  toFirestore: () => {
      throw new Error(`toFirestore is not implemented for ReadAppUserPostLike`)
  }
})
}

/**
 * Provides a reference to a appUserPostLike document for reading.
 * @param appUserPostLikeId - The ID of the appUserPostLike document to read.
 */
export const readAppUserPostLikeDocumentReference = ({
  appUserId,
appUserPostId,
  appUserPostLikeId,
}: {
  appUserId: string
appUserPostId: string,
  appUserPostLikeId: string
}): DocumentReference<ReadAppUserPostLike> =>
    readAppUserPostLikeCollectionReference({
      appUserId,
appUserPostId
    }).doc(appUserPostLikeId);


/**
 * Provides a reference to the appUserPostLikes collection for creating.
 */
export const createAppUserPostLikeCollectionReference = (
  {
  appUserId,
appUserPostId
}: {
  appUserId: string,
appUserPostId: string
}
): CollectionReference<CreateAppUserPostLike>
=> { return db
.collection('appUsers').doc(appUserId).collection('appUserPosts').doc(appUserPostId).collection('appUserPostLikes').withConverter<CreateAppUserPostLike>({
  fromFirestore: () => {
    throw new Error(`fromFirestore is not implemented for CreateAppUserPostLike`)
  },
  toFirestore: (obj: CreateAppUserPostLike): DocumentData => {
      return obj.toJson()
  }
})
}

/**
 * Provides a reference to a appUserPostLike document for creating.
 * @param appUserPostLikeId - The ID of the appUserPostLike document to read.
 */
export const createAppUserPostLikeDocumentReference = ({
  appUserId,
appUserPostId,
  appUserPostLikeId,
}: {
  appUserId: string
appUserPostId: string,
  appUserPostLikeId: string
}): DocumentReference<CreateAppUserPostLike> =>
    createAppUserPostLikeCollectionReference({
      appUserId,
appUserPostId
    }).doc(appUserPostLikeId);


/**
 * Provides a reference to the appUserPostLikes collection for updating.
 */
export const updateAppUserPostLikeCollectionReference = (
  {
  appUserId,
appUserPostId
}: {
  appUserId: string,
appUserPostId: string
}
): CollectionReference<UpdateAppUserPostLike>
=> { return db
.collection('appUsers').doc(appUserId).collection('appUserPosts').doc(appUserPostId).collection('appUserPostLikes').withConverter<UpdateAppUserPostLike>({
  fromFirestore: () => {
    throw new Error(`fromFirestore is not implemented for UpdateAppUserPostLike`)
  },
  toFirestore: (obj: UpdateAppUserPostLike): DocumentData => {
      return obj.toJson()
  }
})
}

/**
 * Provides a reference to a appUserPostLike document for updating.
 * @param appUserPostLikeId - The ID of the appUserPostLike document to read.
 */
export const updateAppUserPostLikeDocumentReference = ({
  appUserId,
appUserPostId,
  appUserPostLikeId,
}: {
  appUserId: string
appUserPostId: string,
  appUserPostLikeId: string
}): DocumentReference<UpdateAppUserPostLike> =>
    updateAppUserPostLikeCollectionReference({
      appUserId,
appUserPostId
    }).doc(appUserPostLikeId);


/**
 * Provides a reference to the appUserPostLikes collection for deleting.
 */
export const deleteAppUserPostLikeCollectionReference = (
  {
  appUserId,
appUserPostId
}: {
  appUserId: string,
appUserPostId: string
}
): CollectionReference<DeleteAppUserPostLike>
=> { return db
.collection('appUsers').doc(appUserId).collection('appUserPosts').doc(appUserPostId).collection('appUserPostLikes').withConverter<DeleteAppUserPostLike>({
  fromFirestore: () => {
    throw new Error(`fromFirestore is not implemented for DeleteAppUserPostLike`)
  },
  toFirestore: (): DocumentData => {
    throw new Error(`toFirestore is not implemented for DeleteAppUserPostLike`)
  }
})
}

/**
 * Provides a reference to a appUserPostLike document for deleting.
 * @param appUserPostLikeId - The ID of the appUserPostLike document to read.
 */
export const deleteAppUserPostLikeDocumentReference = ({
  appUserId,
appUserPostId,
  appUserPostLikeId,
}: {
  appUserId: string
appUserPostId: string,
  appUserPostLikeId: string
}): DocumentReference<DeleteAppUserPostLike> =>
    deleteAppUserPostLikeCollectionReference({
      appUserId,
appUserPostId
    }).doc(appUserPostLikeId);



/**
 * Manages queries against the appUserPostLikes collection.
 */
export class AppUserPostLikeQuery {
  /**
   * Fetches appUserPostLike documents.
   * @param queryBuilder - Function to modify the query.
   * @param compare - Function to sort the results.
   */
  async fetchDocuments({
      appUserId,
appUserPostId,
      queryBuilder,
      compare
  }: {
      appUserId: string
appUserPostId: string,
      queryBuilder?: (
          query: Query<ReadAppUserPostLike>
      ) => Query<ReadAppUserPostLike>
      compare?: (lhs: ReadAppUserPostLike, rhs: ReadAppUserPostLike) => number
  }): Promise<ReadAppUserPostLike[]> {
      let query: Query<ReadAppUserPostLike> =
          readAppUserPostLikeCollectionReference({appUserId,appUserPostId})
      if (queryBuilder != undefined) {
          query = queryBuilder(query)
      }
      const qs: QuerySnapshot<ReadAppUserPostLike> = await query.get()
      let result = qs.docs.map((qds: QueryDocumentSnapshot<ReadAppUserPostLike>) =>
          qds.data()
      )
      if (compare != undefined) {
          result = result.sort(compare)
      }
      return result
  }

  /**
   * Fetches a specific appUserPostLike document.
   * @param appUserPostLikeId - The ID of the appUserPostLike document to fetch.
   */
  async fetchDocument({
    appUserId,
appUserPostId,
    appUserPostLikeId
  }: {
    appUserId: string
appUserPostId: string,
    appUserPostLikeId: string
  }): Promise<ReadAppUserPostLike | undefined> {
    const ds = await readAppUserPostLikeDocumentReference({
      appUserId,
appUserPostId,
      appUserPostLikeId
    }).get()
    return ds.data()
  }

  /**
   * Adds a appUserPostLike document.
   * @param createAppUserPostLike - The appUserPostLike details to add.
   */
  async add({
    appUserId,
appUserPostId,
    createAppUserPostLike
  }: {
    appUserId: string
appUserPostId: string,
    createAppUserPostLike: CreateAppUserPostLike
  }): Promise<DocumentReference<CreateAppUserPostLike>> {
    return createAppUserPostLikeCollectionReference({appUserId,appUserPostId}).add(createAppUserPostLike)
  }

  /**
   * Sets a appUserPostLike document.
   * @param appUserPostLikeId - The ID of the appUserPostLike document to set.
   * @param createAppUserPostLike - The appUserPostLike details to set.
   * @param options - Options for the set operation.
   */
  async set({
      appUserId,
appUserPostId,
      appUserPostLikeId,
      createAppUserPostLike,
      options
  }: {
      appUserId: string
appUserPostId: string,
      appUserPostLikeId: string
      createAppUserPostLike: CreateAppUserPostLike
      options?: SetOptions
  }): Promise<WriteResult> {
      if (options == undefined) {
          return createAppUserPostLikeDocumentReference({
            appUserId,
appUserPostId,
            appUserPostLikeId
          }).set(createAppUserPostLike)
      } else {
          return createAppUserPostLikeDocumentReference({ 
            appUserId,
appUserPostId,
            appUserPostLikeId 
            }).set(createAppUserPostLike, options)
      }
  }

  /**
   * Updates a specific appUserPostLike document.
   * @param appUserPostLikeId - The ID of the appUserPostLike document to update.
   * @param updateAppUserPostLike - The details for updating the appUserPostLike.
   */
  async update({
    appUserId,
appUserPostId,
    appUserPostLikeId,
    updateAppUserPostLike
  }: {
    appUserId: string
appUserPostId: string,
    appUserPostLikeId: string
    updateAppUserPostLike: UpdateAppUserPostLike
  }): Promise<WriteResult> {
      return updateAppUserPostLikeDocumentReference({ 
        appUserId,
appUserPostId,
        appUserPostLikeId 
      }).update(updateAppUserPostLike.toJson())
  }

  /**
   * Deletes a specific appUserPostLike document.
   * @param appUserPostLikeId - The ID of the appUserPostLike document to delete.
   */
  async delete({
    appUserId,
appUserPostId,
    appUserPostLikeId
  }: {
    appUserId: string
appUserPostId: string,
    appUserPostLikeId: string
  }): Promise<WriteResult> {
      return deleteAppUserPostLikeDocumentReference({ 
        appUserId,
appUserPostId,
        appUserPostLikeId 
      }).delete()
  }
}
